radio/src/gui/colorlcd/view_text.cpp

```diff
@@ -57,236 +42,166 @@ void ViewTextWindow::extractNameSansExt()
 
-      maxPos =
-          (maxLines - maxScreenLines) * (PAGE_LINE_HEIGHT + PAGE_LINE_SPACING);
-      if (maxPos < body.getRect().h) maxPos = body.getRect().h;
+    buffer = (char *)malloc(bufSize);
+    if (buffer)
+    {
+      offset = std::max( int(openFromEnd ? int(info.fsize) - bufSize + 1 : 0), 0);
+      TRACE("info.fsize=%d\tbufSize=%d\toffset=%d", info.fsize, bufSize,  int(info.fsize) - bufSize + 1);
+      if (sdReadTextFileBlock(fullPath.c_str(), bufSize, offset) == FR_OK) {
+        //ProcessTextBlock(buffer);
+        
+        auto obj = window->getLvObj();
+        lv_obj_add_flag(
+            obj, LV_OBJ_FLAG_SCROLLABLE | LV_OBJ_FLAG_SCROLL_WITH_ARROW |
+                     LV_OBJ_FLAG_SCROLL_MOMENTUM | LV_OBJ_FLAG_CLICK_FOCUSABLE);
+        // lv_obj_clear_flag(obj, LV_OBJ_FLAG_CLICKABLE);
+        lv_obj_set_scrollbar_mode(obj, LV_SCROLLBAR_MODE_AUTO);
+        // lvb = obj;
+        auto g = lv_group_get_default();
+        lb = lv_label_create(obj);
+        lv_obj_set_size(lb, lv_pct(100), LV_SIZE_CONTENT);
+        //lv_label_set_long_mode(lb, LV_LABEL_LONG_DOT);
+        lv_obj_set_style_pad_all(lb, lv_dpx(8), 0);
+
+        lv_group_add_obj(g, obj);
+        lv_group_set_editing(g, true);
+        lv_label_set_text_static(lb, buffer);
+        lv_obj_add_event_cb(obj, ViewTextWindow::tv_event, LV_EVENT_SCROLL_END,
+                            nullptr);
+
+        if (openFromEnd)
+          lv_obj_scroll_to_y(obj, LV_COORD_MAX, LV_ANIM_OFF);
+        else
+          lv_obj_scroll_to_y(obj, 0, LV_ANIM_OFF);
+      }
     }
   }
-
-  isInSetup = false;
-  textVerticalOffset = 0;
-  loadOneScreen(openFromEnd ? (maxLines - maxScreenLines) : 0);
-
-  for (i = 0; i < maxScreenLines; i++) {
-    new DynamicText(window, grid.getSlot(), [=]() {
-      std::string str =
-          (lines[i][0]) ? std::string(lines[i]) : std::string(" ");
-      return std::string(str);
-    });
-    grid.nextLine();
-    }
 }
 
-#if defined(HARDWARE_TOUCH)
-bool ViewTextWindow::onTouchSlide(coord_t x, coord_t y, coord_t startX,
-                                  coord_t startY, coord_t slideX,
-                                  coord_t slideY)
+void ViewTextWindow::tv_event(lv_event_t *e)
 {
-  // if (&body == Window::focusWindow) {
-  //   const int step = PAGE_LINE_HEIGHT + PAGE_LINE_SPACING;
-  //   int deltaY = -slideY;
-  //   int lineStep = deltaY / step;
-
-  //   textVerticalOffset += lineStep;
-  //   if (textVerticalOffset < 0) textVerticalOffset = 0;
-
-  //   if (textVerticalOffset > maxLines - maxScreenLines)
-  //     textVerticalOffset = maxLines - maxScreenLines;
-  //   sdReadTextFileBlock(fullPath.c_str(), readLinesCount);
-  // }
-  // return Page::onTouchSlide(x, y, startX, startY, slideX, slideY);
-  return true;
+//  lv_event_code_t code = lv_event_get_code(e);
+  lv_obj_t *obj = lv_event_get_target(e);
+  if (obj) {
+    lv_group_t *g = (lv_group_t *)lv_obj_get_group(obj);
+    if (g) lv_group_set_editing(g, true);
```

Comment:
If you remove CLICK_FOCUSABLE from the form containing the label, you won't need this, as shown by f5f453c3cb2.

Reply:
Thank you! Done.